src/html_support_files/odoc.css

```diff
@@ -754,11 +773,217 @@ td.def-doc *:first-child {
   display: block;
 }
 
-.odoc-toc ul li a:hover {
+.odoc-sidebar ul li a:hover {
   box-shadow: none;
   text-decoration: underline;
 }
 
+.odoc-search {
+  position: sticky;
+  top: 0;
+  height: min-content;
+  background: var(--main-background);
+  width: 100%;
+  padding-top: 1rem;
+  z-index: 1;
+}
+
+
+.odoc-search .search-inner {
+  width: 100%;
+  position: relative;
+  left:0;
+  transition : left 0.3s, transform 0.3s, width 0.3s; 
```

Comment:
The slow transition is problematic on chrome, so I think we can remove it.
However I am not sure that liking something or not is a very good reason to pick the style. The search

Reply:
The search results have a clearly different style than the rest of the page and clearly defined borders and background. I don't think its size can be confusing.

- If more width is needed, the element can be extended to the right. Extending to the left requires more code, a more complex layout and just feels weird.

- The empty space on the left is used by the side bar and can be small on some screens. The search bar could touch the side bar.

- When reading on a small screen, the width of page might change. If not, the level of zoom will have to be changed to fit the search bar.

> However I am not sure that liking something or not is a very good reason to pick the style.

Disagree. That a majority of people dislike a style is a reason to change it. I stated my opinion first but hopefully more people will state their opinions.

Reply:
> The search results have a clearly different style than the rest of the page and clearly defined borders and background. I don't think its size can be confusing.
It was very confusing at one point. Maybe its not anymore because of effort mades in other directions, but the size contributes.

>If more width is needed, the element can be extended to the right.
More width is not needed, its just nice to have, and gives more info to the user. 
I think extending on the right only is very ugly, because of the lack of symmetry. If I saw that in the wild, I would think it was a bug. 

> The search bar could touch the side bar.
That should not be hard to prevent by adding a margin, ie the search results would not be larger if it would not fit.

> When reading on a small screen, the width of page might change. If not, the level of zoom will have to be changed to fit the search bar.
I agree that this style might break on small screen, but as far as I remember, everything break on small screens. Even if it did not, I dont think we should make the style worse on big screens for it to be better on small ones, css offers things to control how things looks on every size of screen, and we should strive for the best in each one.

> Disagree. That a majority of people dislike a style is a reason to change it. I stated my opinion first but hopefully more people will state their opinions.
I wanted to know your reasons, which I believe are in a way more important than your conclusion.

Reply:
I took some screenshots.

1. This is what is currently in the code :
![image](https://github.com/ocaml/odoc/assets/27727887/591139dc-4fbd-4b0a-ba97-da40bacfb7f9)

2. This is with no extra width :
![image](https://github.com/ocaml/odoc/assets/27727887/82bbc889-1277-47a3-971d-19e4fe5cc35e)

It lack a bit of polish, you can see stuff sticking out of sides. Its also quite quite confusing.

3. This is with the same extra width, expect we only take it on the right.
![image](https://github.com/ocaml/odoc/assets/27727887/81b7161f-0bfd-40fa-bd0d-542a62ab83bf)

I maintain this looks like a weird bug, but its not confusing, and gives the extra space.

Reply:
I also think that not everyone's opinion is equal on this type of stuff. I asked Claire (web designer at tarides), and she agreed option 2 was too confusing, and also pointed out that the fact that you could both scroll the results and the background made it even worse.
She also did not like the lack of symmetry in option 3, but thought it was better than option 2.
She proposed to blur everything else :
![image](https://github.com/ocaml/odoc/assets/27727887/57114f01-13e1-42ea-b4ae-06b67c7c33bb)
But I think this could get old quickly if you needed to compare search results with doc on the page.

Reply:
I do not agree that extending to the right looks like a bug. Though I'd prefer that it does not change at all if more space is not needed.

We should not make the situation worse on small screens. The complex CSS needed to accommodate small screens is one more arguments against the changing elements. 

On a light color scheme, there's a bigger contrast between the background of the search results and the page. 

We're seeing on small column of pixel from the page on the left of the search results. I think that looks like a bug as I'd expect the search result to either completely overlap what's under it or to overlap a lot less.

Reply:
Ocaml.org picked option 2 :
![image](https://github.com/ocaml/odoc/assets/27727887/4f72d7d9-0902-46cb-a9c4-ee839cbaa5c6)
But they have some sort of side-bar which reduces the asymmetry, and also makes the search not "bigger" in any way : 
![image](https://github.com/ocaml/odoc/assets/27727887/34a0f492-ad59-4be1-beff-8453179adfdf)

Reply:
> Though I'd prefer that it does not change at all if more space is not needed.

We cannot do this on a "as needed" basis, because then the div would change size when you type or scroll.

> We should not make the situation worse on small screens. 

It does not make it worse because its already completely impossible to use : 
![image](https://github.com/ocaml/odoc/assets/27727887/366b1809-30ee-48d8-9cee-169672ef3fe3)

>  The complex CSS needed to accommodate small screens is one more arguments against the changing elements.

I do not think it would be that complex, its already the only thing that works correctly one phone screens : 
![image](https://github.com/ocaml/odoc/assets/27727887/ba29b047-f74e-4dd5-9f27-b1750df91a51)

Reply:
> We cannot do this on a "as needed" basis, because then the div would change size when you type or scroll.

By "if needed" I mean if the feature is needed at all. I'm not convinced this extra space is useful.

Reply:
Here is what I think about this CSS issue:
- Personally, I like the rendered bar as it is
- I understand that it might not be the most "consensual" design and that enough people don't like "when it moves".
- I don't think that any design proposed by Jules is "technically better". I think all of them will have some weird cases where they don't render ideally, all of them will have to be maintained a bit, and none of them will create unsolvable problem.
- I don't think it's a big problem to change the css to something else later if it turns out not good.

My conclusion is the following: I don't think it's reasonable to block this PR on discussion about the CSS.
@EmileTrotignon has done some great work, with advices from Claire. Maybe it's not not consensual, it's clearly better than nothing.

Once this PR is merged, I would welcome a PR modifying the CSS (a concrete proposition would help compare the two alternatives).